app/src/main/java/io/appium/uiautomator2/handler/CaptureScreenshot.java

```diff
@@ -1,17 +1,44 @@
 package io.appium.uiautomator2.handler;
 
-import android.os.Environment;
+import android.app.UiAutomation;
+import android.graphics.Bitmap;
+import android.support.annotation.Nullable;
+import android.util.Base64;
 
-import java.io.File;
+import java.io.ByteArrayOutputStream;
 
 import io.appium.uiautomator2.handler.request.SafeRequestHandler;
 import io.appium.uiautomator2.http.AppiumResponse;
 import io.appium.uiautomator2.http.IHttpRequest;
+import io.appium.uiautomator2.model.internal.CustomUiDevice;
 import io.appium.uiautomator2.server.WDStatus;
-import io.appium.uiautomator2.utils.Device;
 import io.appium.uiautomator2.utils.Logger;
 
+
 public class CaptureScreenshot extends SafeRequestHandler {
+    static final int SCREENSHOT_COMPRESSION_QUALITY = 80;
+
+    private static final UiAutomation uia = CustomUiDevice.getInstance()
+            .getInstrumentation()
+            .getUiAutomation();
+
+    @Nullable
+    private static String takeScreenshot() {
+        final Bitmap screenshot = uia.takeScreenshot();
+        if (screenshot == null) {
+            return null;
+        }
+        try {
+            final ByteArrayOutputStream stream = new ByteArrayOutputStream();
+            if (!screenshot.compress(Bitmap.CompressFormat.PNG, SCREENSHOT_COMPRESSION_QUALITY,
```

Comment:
`PNG` is lossless format - quality will be ignored

Reply:
true. Will set to 100